lib/ansible/modules/monitoring/zabbix/zabbix_map.py

```diff
@@ -0,0 +1,812 @@
+            'name': self.map_name,
+            'label_type': self._get_label_type_id(self.label_type),
+            'expandproblem': int(self.expand_problem),
+            'highlight': int(self.highlight),
+            'width': self.width,
+            'height': self.height,
+            'selements': self._get_selements(graph, nodes, icon_ids),
+            'links': self._get_links(nodes, edges),
+        }
+        return map_config
+
+    def _get_label_type_id(self, label_type):
+        label_type_ids = {
+            'label': 0,
+            'ip': 1,
+            'name': 2,
+            'status': 3,
+            'nothing': 4,
+            'custom': 5,
+        }
+        try:
+            label_type_id = label_type_ids[label_type]
+        except Exception as e:
+            self._module.fail_json(msg="Failed to find id for label type '%s': %s" % (label_type, e))
+        return label_type_id
+
+    def _get_images_info(self, data, icon_ids):
+        images = [
+            {
+                'dot_tag': 'zbx_image',
+                'zbx_property': 'iconid_off',
+                'mandatory': True
+            },
+            {
+                'dot_tag': 'zbx_image_disabled',
+                'zbx_property': 'iconid_disabled',
+                'mandatory': False
+            },
+            {
+                'dot_tag': 'zbx_image_maintenance',
+                'zbx_property': 'iconid_maintenance',
+                'mandatory': False
+            },
+            {
+                'dot_tag': 'zbx_image_problem',
+                'zbx_property': 'iconid_on',
+                'mandatory': False
+            }
+        ]
+        images_info = {}
+        default_image = self.default_image if self.default_image else sorted(icon_ids.items())[0][0]
+        for image in images:
+            image_name = data.get(image['dot_tag'], None)
+            if not image_name:
+                if image['mandatory']:
+                    image_name = default_image
+                else:
+                    continue
+            image_name = remove_quotes(image_name)
+            if image_name in icon_ids:
+                images_info[image['zbx_property']] = icon_ids[image_name]
+                if not image['mandatory']:
+                    images_info['use_iconmap'] = 0
+            else:
+                self._module.fail_json(msg="Failed to find id for image '%s'" % image_name)
+        return images_info
+
+    def _get_element_type(self, data):
+        types = {
+            'host': 0,
+            'sysmap': 1,
+            'trigger': 2,
+            'group': 3,
+            'image': 4
+        }
+        element_type = {
+            'elementtype': types['image'],
+            # Elementid was mandatory even for image elements until Zabbix 3.4.
+            # Since Zabbix API ignores unknown fields, this is left here for simplicity:
+            # it works for Zabbix <= 3.2 and doesn't break Zabbix 3.4.
+            'elementid': "0",
```

Comment:
Commenting this seems to fix the error for zabbix-server 3.4

Reply:
@D3DeFi, thank you for your response! I have fixed the bug, updated PR description and also checked the module against Zabbix 4.0, which seems to be working as well.